distributed/client.py

```diff
@@ -394,6 +396,16 @@ def __init__(self, address=None, start=True, loop=None, timeout=5,
         self.connection_args = self.security.get_connection_args('client')
         self._connecting_to_scheduler = False
 
+        if loop is None:
+            self._should_close_loop = None
+            if in_ioloop():
+                self.loop = IOLoop.current()
```

Comment:
Normal users yes, although there are also plenty of advanced people doing strange things.  They might want to construct their own systems and manage their own IOLoops.

I have been frequently surprised by what people do and far deep they go into internals.  I would be cautious of any change that removed flexibility.